### PR TITLE
[Mailer] Switching to new config format

### PR DESCRIPTION
**Pull request by ThomasLandauer**
Page: https://symfony.com/doc/5.4/mailer.html

### CODE REVIEW

File: mailer.rst [hunk 1]
```diff
@@ -55,14 +55,10 @@ over SMTP by configuring the DSN in your ``.env`` file (the ``user``,
...
-        return static function (ContainerConfigurator $containerConfigurator): void {
-            $containerConfigurator->extension('framework', [
-                'mailer' => [
-                    'dsn' => env('MAILER_DSN'),
-                ],
-            ]);
+        use Symfony\Config\FrameworkConfig;
+
+        return static function (FrameworkConfig $frameworkConfig): void {
+            $frameworkConfig->mailer()->dsn(env('MAILER_DSN')->string());
```
Comment (94noni) [inline mailer.rst]: what does the `string()` at the end?
sorry I rarely use such config way thus only bare comparing array config vs php function here

Comment (ThomasLandauer) [reply mailer.rst]: It's called an "[Environment Variable Processor](https://symfony.com/doc/current/configuration/env_var_processors.html#built-in-environment-variable-processors)" - I now added it to YAML & XML too.

Comment (94noni) [reply mailer.rst]: ho i see now, know it in other config lang but not php, thx for clarifying 👍🏻

**Comment by javiereguiluz** [comment]
Is this change really necessary? Aren't DSN already strings?

**Comment by ThomasLandauer** [comment]
The important part is the lower part of the PR: Switching from the old array syntax to the newer Configurator syntax.

**Comment by OskarStark** [comment]
The string stuff can be removed 👍

**Comment by ThomasLandauer** [comment]
Well, I thought that casting it to a string (just to make sure) would be a good idea. Don't you think?

**Comment by xabbuh** [comment]
I don't think so. If I don't miss anything, we don't do that anywhere else.

**Comment by ThomasLandauer** [comment]
OK, there you go :-)

**Comment by javiereguiluz** [comment]
Merged! Sorry it took us so long. Thanks Thomas.

**Pull request merged**